Giza olympia fix bundled video creation and nft issuance

dobertRowneySr opened the pull request:
fixes:
- #3107
- #3106 

I also removed all the unused code like `PersonId` and the likes that was not removed this far, in order to have cleaner type aliases

~*Update 2022-02-02*: rebased to current `olympia` . This lead to a larger number of commits
Relevant comits start from : c86b9ddb8029437e854b166797e513b5efd6d703~
Relevant code section are in the `content` module, in particular:
- `NFTIssuanceParameter` type introduced
- `InitTransactionalStatus` type introduced
- `ensure_valid_init_transactional_status` guard added
- `create_video` & `issue_nft` extrinsic argument modified

Commit:
added issuance type

Commit:
added from implementation

Commit:
fixed mocks & buy_nft tests

Commit:
fixed cancel_buy_now tests

Commit:
fixed cancel nft auction tests

Commit:
fixed cancel offer tests

Commit:
cancel nft open auction bid

Commit:
updated claim won english auction tests

Commit:
Issue NFT tests updated

Commit:
updated make bid tests

Commit:
updated offer nft tests

Commit:
updated open auction winner tests

Commit:
updated sell nft tests

Commit:
updated sling back nft tests

Commit:
updated start nft auction

Commit:
small fix to tests

Commit:
updated accept incoming offer tests

Commit:
updated fixed accept incoming offer tests

Commit:
added accept incoming offer tests

Commit:
added params validation before setting the status

Commit:
updated InitTransactionalStatus

Commit:
adedd checks for auction parameters

Commit:
added tests to ensure issue_nft init transactional status guard

Commit:
formatting

Commit:
nft can now be issued at video creation

Commit:
removed unused code & added nft construction routine

Commit:
updated tests

Commit:
updated runtime code

shamil-gadelshin commented on runtime/src/constants.rs:
```diff
@@ -245,7 +245,6 @@ lazy_static! {
         ].to_vec()),
         // Bounty
         (BountyLockId::get(), [
-            InvitedMemberLockId::get(),
```
Why did you remove that?

dobertRowneySr replied:
I am not sure why github as that. It is marked as outdated, and the line is present on my code

shamil-gadelshin commented on scripts/generate-weights.sh:
```diff
@@ -68,4 +68,4 @@ benchmark membership
 benchmark bounty
 benchmark blog
 benchmark joystream_utility
-benchmark storage_v2
```
Comment the line, please. It produces an error.

dobertRowneySr replied:
fixed in 8184b6caecc3bf303b9afd0d9f1ee52464ef0597

dobertRowneySr committed:
removed membership Trait from benchmarks

dobertRowneySr committed:
fixed tests' Extbuilder

dobertRowneySr committed:
fixed runtime api

shamil-gadelshin commented on runtime-modules/content/src/lib.rs:
```diff
@@ -2256,6 +2100,59 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    pub fn ensure_valid_init_transactional_status(
```
Do we need this method as public?

dobertRowneySr replied:
fixed in 92a405cf94a779685a5ed4f648210345e00a75fa

shamil-gadelshin commented on runtime-modules/content/src/lib.rs:
```diff
@@ -2256,6 +2100,59 @@ impl<T: Trait> Module<T> {
+        }
+    }
+
+    pub fn construct_owned_nft(
```
Do we need this method as public?

dobertRowneySr replied:
fixed in 92a405cf94a779685a5ed4f648210345e00a75fa

shamil-gadelshin commented on runtime-modules/content/src/lib.rs:
```diff
@@ -2256,6 +2100,59 @@ impl<T: Trait> Module<T> {
+    pub fn ensure_valid_init_transactional_status(
+        init_status: &InitTransactionalStatus<T>,
+    ) -> Result<
+        TransactionalStatus<
```
It makes sense to introduce an alias for such a complex type.

dobertRowneySr replied:
fixed in afcde3072cf9c430da415bf088c3e96196d7f02b

shamil-gadelshin reviewed (changes requested):
The underlying issues look addressed. However, there are multiple issues with the code:
 - no doc-comments for public types, extrinsics, and methods
 - please, remove `allow all` clippy warning statement and fix all clippy issues in the pallet
 - found this line in the code: `// TODO: Solve #now` 
 - rebase seems unsuccessful: 1) there is still unrelated code in the PR (removed empty extrinsics), 2) `generate_weights.sh` script fails now (incorrect benchmark name)
 - there are public methods with no external usage (should be private)

dobertRowneySr committed:
commented storage benchmark

dobertRowneySr committed:
made methods private

dobertRowneySr committed:
removed allow all

dobertRowneySr committed:
added alias for TransactionalStatus

dobertRowneySr committed:
fixed test for the new TransactionalStatus alias

dobertRowneySr committed:
removed obsolete comment and added doc comments to nft/types

dobertRowneySr committed:
doc-commented helper functions

dobertRowneySr commented:
> The underlying issues look addressed. However, there are multiple issues with the code:
> 
>     * no doc-comments for public types, extrinsics, and methods
> 
>     * please, remove `allow all` clippy warning statement and fix all clippy issues in the pallet
> 
>     * found this line in the code: `// TODO: Solve #now`
> 
>     * rebase seems unsuccessful: 1) there is still unrelated code in the PR (removed empty extrinsics), 2) `generate_weights.sh` script fails now (incorrect benchmark name)
> 
>     * there are public methods with no external usage (should be private)

- the `TODO` was an old reference to myself that I solved and forgot to remove
- benchmark should be fixed
- removed the cargo directive 
- added doc-comments on types and helper functions added

I could locate the `removed_empty_extrinsic` reference in the Github commit tab

dobertRowneySr committed:
fixed format errors

shamil-gadelshin commented on runtime-modules/content/src/lib.rs:
```diff
@@ -1031,54 +962,6 @@ decl_module! {
 
             Self::deposit_event(RawEvent::VideoCategoryDeleted(actor, category_id));
         }
-
```
Isn't it a rebase artifact?

dobertRowneySr replied:
I am not following. The BASE commit in olympia has this code. In this PR that I am opening I removed it

shamil-gadelshin replied:
Ok. So, it's part of the PR. I was expecting only the NFT-related code.

dobertRowneySr committed:
clippy warnings fixed

dobertRowneySr committed:
updated types

dobertRowneySr committed:
generated types

shamil-gadelshin reviewed (approved):
LGTM
